tmva/sofie/inc/TMVA/ROperator_Softmax.hxx

```diff
@@ -58,199 +58,108 @@ public:
          throw std::runtime_error("TMVA SOFIE Operator Softmax called to Generate without being initialized first");
       }
       std::stringstream out;
-      int size = fShape.size();
-      auto length = ConvertDimShapeToLength(fShape);
-      auto stride = UTILITY::ComputeStrideFromShape(fShape);
+      size_t size = fShape.size();
+      auto length_str = ConvertDimShapeToLength(fShape);
+      size_t length = std::stoul(length_str);
       int axis = fAttrAxis < 0 ? size + fAttrAxis : fAttrAxis;
-      out << "\n" << SP << "//------ SOFTMAX - " << size << "  " << length << "  " << axis << "\n";
-      // use safe numerically implementation by subtracting max of tensor
-      if (size == 1) {
-         out << SP << fType << " vmax = tensor_" << fNX << "[0];\n";
-         out << SP << "for (size_t i = 1; i < " << length << " ; i++){\n";
-         out << SP << SP << "if (tensor_" << fNX << "[i] > vmax) vmax = tensor_" << fNX << "[i];\n";
-         out << SP << "}\n";
-         out << SP << fType << " sum = 0.0;\n";
-         out << SP << "for (size_t i = 0; i < " << length << " ; i++){\n";
-         out << SP << SP << "tensor_" << fNY << "[i] = std::exp(tensor_" << fNX << "[i] - vmax);\n";
-         out << SP << SP << "sum += tensor_" << fNY << "[i];\n";
-         out << SP << "}\n";
-         out << SP << "for (size_t i = 0; i < " << length << " ; i++){\n";
-         out << SP << SP << "tensor_" << fNY << "[i] /= sum;\n";
+
+      // Check if this is the special case where memory is contiguous.
+      if (axis == static_cast<int>(size - 1)) {
+         size_t axis_size = std::stoul(fShape[axis].GetVal());
+         size_t num_rows = length / axis_size;
```

Comment:
In case of length is not an integer , this could be written as following: 
```
std::string axis_size = fShape[axis].GetVal());
std::string num_rows; 
if (IsInteger(length) && IsInteger(axis_size)) { 
  num_rows = std::to_string( std::stoul(length_str)/ std::stoul(fShape[axis].GetVal()));
} else  {
  num_rows = "(" + length_str + ") /" + fShape[axis].GetVal():
}